crapssim/bet.py

```diff
@@ -473,7 +516,37 @@ def __init__(self, bet_amount: float):
 class Hard10(Hardway):
     def __init__(self, bet_amount: float):
         super().__init__(bet_amount)
-        self.name: str ="Hard10"
+        self.name: str = "Hard10"
         self.winning_result: list[int | None] = [5, 5]
         self.number: int = 10
         self.payoutratio: int = 7
+
+
+class Fire(Bet):
+    def __init__(self, bet_amount: float):
+        super().__init__(bet_amount)
+        self.name: str = "Fire"
+        self.bet_amount: float = bet_amount
+        self.points_made: list[int] = []
+        self.current_point: int | None = None
+
+    def _update_bet(self, table_object: "Table", dice_object: Dice) -> tuple[str | None, float, bool]:
+        status, win_amount, remove = None, 0.0, False
+        if self.current_point is None and dice_object.total in (4, 5, 6, 8, 9, 10):
+            self.current_point = dice_object.total
+        elif self.current_point is not None and self.current_point == dice_object.total:
+            if self.current_point not in self.points_made:
+                self.points_made = list(set(self.points_made + [dice_object.total]))
+                if len(self.points_made) == 4:
+                    status, win_amount, remove = 'win', 25 * self.bet_amount, False
+                elif len(self.points_made) == 5:
+                    status, win_amount, remove = 'win', 250 * self.bet_amount, False
+                elif len(self.points_made) == 6:
+                    status, win_amount, remove = 'win', 1000 * self.bet_amount, True

```

Comment:
Corrected with 6e6c8f5

Reply:
Yeah, I do like having the pay tables adjustable.

I think it would be nice if we were consistent with where the overridden pay tables are. Right now most of them are on the bet, however we also do have table.payouts which includes some of the field payouts. 

It wouldn't be hard to make the payouts for the field as a parameter for Field as well, but In my mind I think that the payouts do belong on the table. I just don't quite know yet how to best go about doing that.

Reply:
Given that I think the current payouts are the most common, and that fire bet isn't super popular, let's go with the current implementation. 

I'll open a separate issue on how to consistently deal with pay tables, and we can continue the conversation there.

Reply:
Track future improvement in #13